dts/arm/st/l5/stm32l5.dtsi

```diff
@@ -273,6 +273,43 @@
 			label = "LPTIM_1";
 		};
 
+		dma1: dma@40020000 {
+			compatible = "st,stm32-dma-v2";
+			#dma-cells = <4>;
+			reg = <0x40020000 0x400>;
+			interrupts = <29 0 30 0 31 0 32 0 33 0 34 0 35 0 36 0>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB1 0x1>;
+			dma-requests = <8>;
+			dma-offset = <0>;
+			status = "disabled";
+			label = "DMA_1";
+		};
+
+		dma2: dma@40020400 {
+			compatible = "st,stm32-dma-v2";
+			#dma-cells = <4>;
+			reg = <0x40020400 0x400>;
+			interrupts = <80 0 81 0 82 0 83 0 84 0 85 0 86 0 87 0>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB1 0x2>;
+			dma-requests = <8>;
+			dma-offset = <8>;
+			status = "disabled";
+			label = "DMA_2";
+		};
+
+		dmamux1: dmamux@40020800 {
+			compatible = "st,stm32-dmamux";
+			#dma-cells = <4>;
+			reg = <0x40020800 0x400>;
+			interrupts = <27 0>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB1 0x4>;
+			dma-channels = <16>;
+			dma-generators = <4>;
+			dma-requests= <90>;
```

Comment:
In RM0438 Rev 6, dma there is up to 94 requests

Reply:
> In RM0438 Rev 6, dma there is up to 94 requests

RM0438 Rev **7** says 90 requests.

Reply:
Correct, _but_ the count is actually dma-generators + dma-requests = 94, the definition relies on table 96 of the RM0438
(not on list of table 97)

Reply:
This needs to be corrected in stm32l4r5.dtsi & stm32wb.dtsi. Will raise a PR on this.

Reply:
ok, so what is the right value here finally ?

Reply:
> ok, so what is the right value here finally ?

90 is the right value for L5. Other series are having the right values, no change needed.

Reply:
I am still on 94
![image](https://user-images.githubusercontent.com/50730894/124439574-dfc5de00-dd79-11eb-8340-6e9b6ab1cf16.png)

Reply:
From https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/dma/dmamux_stm32.c#L304:L311, gen_nb gets the dma generators and req_nb gets the dma requests from the DT.

```
const struct dmamux_stm32_config dmamux_stm32_config_##index = {        \
        DMAMUX_CLOCK_INIT(index)                                        \
        .base = DT_INST_REG_ADDR(index),                                \
        .channel_nb = DT_INST_PROP(index, dma_channels),                \
        .gen_nb = DT_INST_PROP(index, dma_generators),                  \
        .req_nb = DT_INST_PROP(index, dma_requests),                    \
        .mux_channels = dmamux_stm32_channels_##index,                  \
};
```

Below https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/dma/dmamux_stm32.c#L43:L53 is the definition of gen_nb and req_nb:

```
/* this is the configuration of the dmamux IP */
struct dmamux_stm32_config {
#if DT_INST_NODE_HAS_PROP(0, clocks)
        struct stm32_pclken pclken;
#endif
        uint32_t base;
        uint8_t channel_nb;     /* total nb of channels */
        uint8_t gen_nb; /* total nb of Request generator */
        uint8_t req_nb; /* total nb of Peripheral Request inputs */
        const struct dmamux_stm32_channel *mux_channels;
};
```

Referring RM0438 Rev7 stm32L562 reference manual,
![image](https://user-images.githubusercontent.com/78502494/124506502-68c82e00-dde9-11eb-9005-8aa065eda9bb.png)

In the below code, the max request Id is calculated by adding req_nb (90 in case of L5) + gen_nb (4 in case of L5).

```

int dmamux_stm32_configure(const struct device *dev, uint32_t id,
                                struct dma_config *config)
{
        /* device is the dmamux, id is the dmamux channel from 0 */
        const struct dmamux_stm32_config *dev_config = dev->config;

        /*
         * request line ID for this mux channel is stored
         * in the dma_slot parameter
         */
        int request_id = config->dma_slot;

        if (request_id > dev_config->req_nb + dev_config->gen_nb) {
                LOG_ERR("request ID %d is not valid.", request_id);
                return -EINVAL;
        }
```

Pls let me know your thoughts on this.

Reply:
Ok I agree. Thanks for the explanation.